src/ESPAsyncWebServer.h

```diff
@@ -191,7 +191,7 @@ class AsyncWebServerRequest {
     AsyncWebServer* _server;
     AsyncWebHandler* _handler;
     AsyncWebServerResponse* _response;
-    std::vector<String> _interestingHeaders;
+    std::list<String> _interestingHeaders;
```

Comment:
Will speed up request processing by avoiding array reallocations at the expense of taking a little more memory  because adding interesting headers is done for each entering request